daemon/daemon.go

```diff
@@ -154,11 +154,11 @@ func (daemon *Daemon) containerRoot(id string) string {
 // Load reads the contents of a container from disk
 // This is typically done at startup.
 func (daemon *Daemon) load(id string) (*Container, error) {
-	container := &Container{
-		root:         daemon.containerRoot(id),
-		State:        NewState(),
-		execCommands: newExecStore(),
-	}
+	container := &Container{}
+	container.root = daemon.containerRoot(id)
+	container.State = NewState()
+	container.execCommands = newExecStore()
```

Comment:
I agree, but there's a good reason. When I split out the Container structure into common and platform specific parts, everything is good in go except for the constructor "container:=&Container{field: value, etc}.". Accessors are fine, but the constructor will fail compilation. Hence the need to rework it to be field by field. Sucks, I know. As far as I can tell, it's a limitation in go.